### DISCUSSION TITLE
Additional options

### DISCUSSION THREAD

**Discussion by kienns1996** [iseahound/ImagePut#42] [Chat]
is it possible to add option between 'user32\PrintWindow' and 'BitBlt' when creating bitmap in from_window()? BitBlt is about 10 times faster than PrintWindow. Additionally, CreateCompatibleDC is also about 10% faster than CreateDIBSection

**Comment by iseahound** [comment]
Addressing only BitBlt:

I've offfered an option to use the format `ImagePutWindow([0,0,,, "ahk_class notepad"])` which exposes the `x`,`y`,`w`,`h` parameters. Does this help?

**Reply by kienns1996** [reply]
I mean this, is there another way to create a BitmapBuffer to use PixelSearch from a hwnd without going through it?

  `; Print the window onto the hBitmap using an undocumented flag. https://stackoverflow.com/a/40042587`
  `DllCall("user32\PrintWindow", "ptr", image, "ptr", hdc, "uint", 0x3) ; PW_RENDERFULLCONTENT | PW_CLIENTONLY`
 `; Additional info on how this is implemented: https://www.reddit.com/r/windows/comments/8ffr56/altprintscreen/`

**Reply by iseahound** [reply]
Are you asking me to change the syntax? I'm not sure what you're asking. 

    pic := ImagePutBuffer({screenshot: "A"})
